File: eden/code-samples/eden.c
#include <stdlib.h>

struct Island {
  char *name;
  int length;
  int width;
  char *features[2];
};

int _calcArea(struct Island *i) { return i->length * i->width; }

int get_area(struct Island *i) { return _calcArea(i); }

int main() {
  struct Island i;

  /* Eden is full of beautiful islands, one of them is named Elysian */
  i.name = malloc(7 * sizeof(char));
  i.name = "Elysian";
  i.length = 300;
  i.width = 350;
  i.features[0] = "Sunsets";
  i.features[1] = "Beaches";

  get_area(&i);

  return 0;
}

File: eden/code-samples/eden.php
<?php
/*
 * Eden is full of beautiful islands...
 */
class Island {
  public string $name;
  public int $length;
  public int $width;
  public array $features;

  public function __construct(array $props) {
    $this->name = $props['name'];
    $this->length = $props['length'];
    $this->width = $props['width'];
    $this->features = $props['features'];
  }

  public function area(): int {
    return $this->calcArea();
  }

  private function calcArea(): int {
    return $this->length * $this->width;
  }
}

// ...one of them is named Elysian
$Elysian = new Island([
  'name' => 'Elysian',
  'length' => 300,
  'width' => 350,
  'features' => ['Sunsets', 'Beaches']
]);
?>
